src/logger/src/logger.rs

```diff
@@ -480,191 +432,262 @@ impl fmt::Display for LoggerError {
+            "[TEST-INSTANCE-ID:INFO:logger.rs:0] info\n",
         );
+        validate_log(Box::new(&mut reader_2), "");
+    }
 
-        assert_eq!(
-            format!("{}", LoggerError::IsInitializing),
-            "The logger is initializing. Can't perform the requested action right now."
+    #[test]
+    fn test_create_prefix() {
+        let logger = Logger::mock_new();
+        let mut reader = logger.mock_init();
+
+        // test with empty instance id
+        logger.set_instance_id("".to_string());
+
+        // check that the prefix is empty when all flags are false
+        logger
+            .set_include_level(false)
+            .set_include_origin(false, true);
+        logger.mock_log(Level::Info, "msg");
+        validate_log(Box::new(&mut reader), "[] msg\n");
+
+        // check that the prefix is correctly shown when all flags are true
+        logger
+            .set_include_level(true)
+            .set_include_origin(true, true);
+        logger.mock_log(Level::Info, "msg");
+        validate_log(Box::new(&mut reader), "[INFO:logger.rs:0] msg\n");
+
+        // check show_line_numbers=false
+        logger
+            .set_include_level(true)
+            .set_include_origin(true, false);
+        logger.mock_log(Level::Debug, "msg");
+        validate_log(Box::new(&mut reader), "[DEBUG:logger.rs] msg\n");
+
+        // check show_file_path=false
+        logger
+            .set_include_level(true)
+            .set_include_origin(false, true);
+        logger.mock_log(Level::Error, "msg");
+        validate_log(Box::new(&mut reader), "[ERROR] msg\n");
+
+        // check show_level=false
+        logger
+            .set_include_level(false)
+            .set_include_origin(true, true);
+        logger.mock_log(Level::Info, "msg");
+        validate_log(Box::new(&mut reader), "[logger.rs:0] msg\n");
+
+        // test with a mock instance id
+        logger.set_instance_id(TEST_INSTANCE_ID.to_string());
+
+        // check that the prefix is empty when all flags are false
+        logger
+            .set_include_level(false)
+            .set_include_origin(false, false);
+        logger.mock_log(Level::Info, "msg");
+        validate_log(Box::new(&mut reader), "[TEST-INSTANCE-ID] msg\n");
+
+        // check that the prefix is correctly shown when all flags are true
+        logger
+            .set_include_level(true)
+            .set_include_origin(true, true);
+        logger.mock_log(Level::Warn, "msg");
+        validate_log(
+            Box::new(&mut reader),
+            "[TEST-INSTANCE-ID:WARN:logger.rs:0] msg\n",
         );
+    }
+
+    #[test]
+    fn test_static_logger() {
+        log::set_max_level(log::LevelFilter::Info);
+        LOGGER.set_instance_id(TEST_INSTANCE_ID.to_string());
+
+        let mut reader = LOGGER.mock_init();
+
+        info!("info");
+        validate_log(Box::new(&mut reader), "info\n");
+    }
+
+    #[test]
+    fn test_error_messages() {
```

Comment:
```suggestion
    fn test_error_message() {
```

Reply:
I don't have strong preferences here, I can change the name of the function, but I would keep the plural. Error is an enum that can hold multiple errors. It just happens that we have only one.